core/variable_binary_arithmetic.cpp

```diff
@@ -113,6 +113,48 @@ Variable &Variable::operator/=(const VariableConstProxy &other) & {
   return divide_equals(*this, other);
 }
 
+template <class T1, class T2> T1 &or_equals(T1 &variable, const T2 &other) {
+  transform_in_place<pair_self_t<bool>>(
+      variable, other,
+      overloaded{[](Bool &var_, const Bool &other_) { var_ |= other_; },
+                 [](auto & /*unused*/, const auto & /*unused*/) {
+                   throw std::runtime_error("This operation can only be "
+                                            "performed on containers with "
+                                            "boolean data type.");
+                 },
+                 [](units::Unit &varUnit, const units::Unit &otherUnit) {
+                   expect::equals(varUnit, units::dimensionless);
+                   expect::equals(otherUnit, units::dimensionless);
+                 }});
+
+  return variable;
+}
+
+template <class T1, class T2> Variable or_op(const T1 &a, const T2 &b) {
+  return transform<pair_self_t<bool>>(
+      a, b,
+      overloaded{[](const Bool &var_, const Bool &other_) -> Bool {
+                   return var_ | other_;
+                 },
+                 [](const auto & /*unused*/, const auto & /*unused*/) {
+                   throw std::runtime_error("This operation can only be "
+                                            "performed on containers with "
+                                            "boolean data type.");
+                 },
```

Comment:
@DTasev Could this be the reason for you clang builds failing. Can you remove this overload and just use `auto` instead of `Bool`? There is no need for hand-coding the exception here, transform will throw anyway before you get there since you only support `bool` inputs.

Reply:
It is quite interesting. Adding `-> Bool`, to the lambda that throws, fixes the compiler error.. However, then all the Binary OP tests fail as clang does not seem to properly resolve a Variable of bools to `Bool`, and instead goes to the `auto` lambda every time.

Removing the `auto` overload seems to fail to compile on both compilers

Reply:
@DTasev, put `transform<pair_self_t<Bool>>` instead of  `transform<pair_self_t<bool>>`, `transform` operates with `Variable` internals, which can only be `Bool` (not `bool`) and the overloads are not supporting implicit conversions.

Reply:
Works here: https://github.com/scipp/scipp/blob/ac8b56950d03bc80779edfa415e9a21018707a70/core/test/transform_test.cpp#L142-L147. You should not use `Bool` anywhere on the API.

Reply:
> and instead goes to the `auto` lambda every time (even though the type is a `Bool`

Why overloads?

- Remove the overload that throws.
- Replace `Bool` by `auto`.

Reply:
> > and instead goes to the `auto` lambda every time (even though the type is a `Bool`
> 
> Why overloads?
> 
>     * Remove the overload that throws.
> 
>     * Replace `Bool` by `auto`.



> @DTasev, put `transform<pair_self_t<Bool>>` instead of `transform<pair_self_t<bool>>`, `transform` operates with `Variable` internals, which can only be `Bool` (not `bool`) and the overloads are not supporting implicit conversions.

My fail, `transform` utilizes `underlying_type` internally.

Reply:
Removing the `throw` lambda and making the `Bool, Bool -> Bool` one be `auto, auto -> bool` works. I, too, wonder why clang can't resolve to Bool, Bool though